lib/pos_blockchain.go

```diff
@@ -481,9 +482,164 @@ func (bc *Blockchain) pruneUncommittedBlocks(desoBlock *MsgDeSoBlock) error {
 // commitGrandparents commits the grandparent of the block if possible.
 // Specifically, this updates the CommittedBlockStatus of its grandparent
 // and flushes the view after connecting the grandparent block to the DB.
-func (bc *Blockchain) commitGrandparents(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me.
-	return errors.New("IMPLEMENT ME")
+func (bc *Blockchain) commitGrandparents() error {
+	currentBlock := bc.GetBestChainTip()
+	var canCommit bool
+	var blockToCommit *BlockHash
+	for !canCommit {
+		// If the grandparent block is committed, then we can commit it.
+		// Otherwise, we can't commit it.
+		var committedBlockSeen bool
+		blockToCommit, canCommit, committedBlockSeen = bc.canCommitGrandparent(currentBlock)
+		// If we've seen a committed block, then we can't commit anything new.
+		if committedBlockSeen {
+			return nil
+		}
+		if !canCommit {
+			var exists bool
+			currentBlock, exists = bc.bestChainMap[*currentBlock.Header.PrevBlockHash]
+			if !exists {
+				return errors.Errorf("commitGrandparents: Block %v not found in best chain map", currentBlock.Hash.String())
+			}
+		}
+	}
```

Comment:
Why do we need to loop over the ancestors of the current block here? If I understand this loop correctly, it's intended to find the highest block that can be committed.

It should be fine to call `bc.canCommitGrandparent(currentBlock)` on just the current tip. If the grandparent can be committed, then you can run the commit rule on the grandparent. It shouldn't be possible for the grandparent to not be committable, while another ancestor is committable. 
```suggestion
	blockToCommit, canCommit, _ := bc.canCommitGrandparent(currentBlock)
	if !canCommit {
		return nil
	}	
```

Reply:
Great! okay I was unsure of this assumption.